src/components/ChatInput/ChatInput.js

```diff
@@ -16,6 +16,11 @@ const ChatInput = () => {
   const inputRef = useRef(null);
   const messageRef = useRef();
   const [disableButton, setDisableButton] = useState(true);
+  const [members, setMembers] = useState({});
+  const [filteredMembers, setFilteredMembers] = useState([]);
```

Comment:
having a filter function inside zustand state which performs the logic and returns the filteredState will be a good idea, what do you think?

Reply:
Yes will do it. Thanks

Reply:
But why should we do this?

Reply:
to replace this logic from component
```
                    members.filter(
                      (member) =>
                        member.name
                          .toLowerCase()
                          .includes(message.substring(c + 1).toLowerCase()) ||
                        member.username
                          .toLowerCase()
                          .includes(message.substring(c + 1).toLowerCase())
                    )
```
else moving it to a function on its own `searchToMentionUser` will be a good idea. What do you think?